pytorch_lightning/trainer/training_loop.py

```diff
@@ -504,6 +505,9 @@ def run_training_epoch(self):
         # epoch end hook
         self.run_on_epoch_end_hook(model)
 
+        # increate global step by one to progress to the next epoch
+        self.global_step += 1
```

Comment:
I think you also need to do do self.increment_accumulated_grad_global_step() here, otherwise the total_train_batch_idx does not get updated. I think that's why the tests are failing.

Reply:
I made the change and it works as expected. So I have changed it. 

However, the tests are still failing. Actually, I think I am confused about the test. I ran the local test on my pull request and got this total summary result at the end:
<img width="786" alt="Screen Shot 2020-07-03 at 12 40 46 AM" src="https://user-images.githubusercontent.com/33330101/86444629-e254aa00-bcc5-11ea-9c69-253b481b49f2.png">

I also ran the test over the original Pytorch Lightning master branch and also got this total summary result at the end:
<img width="756" alt="Screen Shot 2020-07-03 at 12 34 03 AM" src="https://user-images.githubusercontent.com/33330101/86444021-fcda5380-bcc4-11ea-8588-a09edb94f565.png">

Am I making a mistake somewhere? Should I have based my branch off on something other than the master branch?